### ISSUE TITLE
`DiffieHellmanSharedSecret::shared_secret` should output a secret type

### ISSUE THREAD

**Issue by sdbondi** [tari-project/tari-crypto#128]
`DiffieHellmanSharedSecret::shared_secret` should return a secret type, `impl SecretKey` or perhaps a concrete `EcdhSecret` type that implements zeroize.

Other considerations are that the secret derivation is not complete and should be passed through a `kdf`.

For example:
```rust
/// This trait provides a common mechanism to calculate a shared secret using the private and public key of two parties
pub trait DiffieHellmanSharedSecret: ByteArray + Clone + PartialEq + Eq + Add<Output = Self> + Default {
    /// The type of public key
    type PK: PublicKey;
    /// Generate a shared secret from one party's private key and another party's public key
    fn shared_secret<K: DerivedKeyDomain>(k: &<Self::PK as PublicKey>::K, pk: &Self::PK, kdf: K) -> SecretKey;
}
```

**Comment by AaronFeickert** [comment]
With the change in [PR 137](https://github.com/tari-project/tari-crypto/pull/137), could we simply have `shared_secret` return a `Zeroizing` wrapper around a public key, and add any necessary `Zeroize` trait bounds? Then it'll be guaranteed to zeroize on drop.

**Comment by AaronFeickert** [comment]
[PR 137](https://github.com/tari-project/tari-crypto/pull/137) now fixes this with a new generic type that implements `Zeroize` and zeroizes on drop.

**Comment by sdbondi** [comment]
Closed by #137
